hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1023,15 +1028,21 @@ private long sleepWithBusyWait(long timeoutInMillis, long sleep) {
 
     @Override
     public boolean isMemberStateSafe() {
+        return getMemberState() == SAFE;
+    }
+
+    public InternalPartitionServiceState getMemberState() {
         if (hasOnGoingMigrationLocal()) {
-            return false;
+            return MIGRATION_LOCAL;
         }
+
         if (!node.isMaster()) {
             if (hasOnGoingMigrationMaster(Level.OFF)) {
-                return false;
+                return MIGRATION_ON_MASTER;
             }
         }
-        return isReplicaInSyncState();
+
+        return isReplicaInSyncState() ? SAFE : REPLICA_NOT_SYNC;
```

Comment:
I didn't put it to InternalPartitionService on purpose. Those methods should be used only for testing.